src/Persistence/Yaml/ControlFormatter.cs

```diff
@@ -20,7 +20,7 @@ public static class ControlFormatter
         _ = control ?? throw new ArgumentNullException(nameof(control));
 
         var childrenToRemove = (control.Children ?? Enumerable.Empty<Control>()).Where(c => c.Template.AddPropertiesToParent).ToList();
-        var propertiesToMerge = childrenToRemove.SelectMany(c => c.Properties).ToList();
+        var propertiesToMerge = childrenToRemove.SelectMany(c => c.Properties.Where(p => p.Key != PropertyNames.ZIndex)).ToList();
```

Comment:
This is already done below on line 40, and the ZIndex still needs to be present for the sorting code on lines 32-35 prior to that removal.

Reply:
or wait, this is the ZIndex of the children with AddPropertiesToParent flag, not this current control's ZIndex.   
Nevermind.